### DISCUSSION TITLE
[@types/react] CSSProperties has no support for css variables

### DISCUSSION THREAD

**Discussion by conlanpatrek** [DefinitelyTyped/DefinitelyTyped#68113] [Issues with a @types package] [answered]
The typing on `CSSProperties` isn't quite correct. React supports css variables by passing them in as `style={{'--my-var-name': '#f00ba7'}}`. When attempting to do this with typescript the compiler throws, because `--my-var-name` isn't part of the `CSSProperties` type definition.

Could there be an additional allowance for this? Is it possible in the type system to allow arbitrary strings as keys IF the key starts with `--`?

Currently the work around is to just cast your cssvariables dict to `CSSVariables` on usage. Would prefer an in-type way to express this.

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
> Is it possible in the type system to allow arbitrary strings as keys IF the key starts with --?

```
[key: `--${string}`]: string
```
should work

**Comment by Semigradsky** [comment, marked as answer]
@conlanpatrek Looks like it is "a feature, not a bug" - see https://github.com/frenic/csstype#what-should-i-do-when-i-get-type-errors
and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/24911

**Reply by conlanpatrek** [reply]
Yeah that seems correct, though frustrating. Thank you bud.

**Reply by merlinaudio** [reply]
Seems like just adding this .d.ts file works. I named it `css.d.ts` like the author suggests, but the name is irrelevant

```ts
// css.d.ts
import type * as CSS from "csstype";

declare module "csstype" {
	interface Properties {
		[index: `--${string}`]: any; // Allow any CSS Custom Properties
	}
}
```
